sequra/database/models.py

```diff
@@ -0,0 +1,118 @@
+import decimal
+
+from sqlalchemy import extract
+from sqlalchemy.orm import validates
+
+from sequra.database import db
+
+
+@validates('email')
+def validate_email(key, address):
+    """ Further work, integration with online email verification service"""
+    assert '@' in address
+    return address
+
+
+class Disbursement(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    _amount = db.Column("amount", db.String(50))
+    week = db.Column(db.Integer)
+    year = db.Column(db.Integer)
+
+    merchant_id = db.Column(db.Integer, db.ForeignKey('merchant.id'), nullable=False)
+
+    merchant = db.relationship('Merchant', backref=db.backref('disbursements', lazy='dynamic'))
+    db.UniqueConstraint('week', 'week', name='unique_disbursement_week_year')
+
+    @property
+    def amount(self):
+        """ SQLite3 does not work with decimal, the built-in representation suitable for currency
+        Can be extracted as a type following the 3 times rule. """
+        return decimal.Decimal(self._amount)
+
+    @amount.setter
+    def amount(self, amount: decimal.Decimal):
+        self._amount = str(amount)
+
+    def save(self, year, week, merchant):
+        self.merchant_id = merchant.id
+        self.week = week
+        self.year = year
+        # self.amount = self.calculate_amount(merchant_id=self.merchant_id, week=self.week, year=self.year)
+
+        db.session.add(self)
+        db.session.commit()
+
+    def asdict(self):
+        return {'amount': float(self.amount), 'week': self.week, 'year': self.year, 'merchant': self.merchant.name}
+
+    @staticmethod
+    def calculate_amount(merchant_id, week, year):
+        orders = Order.amounts_by_merchant_in_week(merchant_id=merchant_id, week=week, year=year)
+        disbursed_amount = decimal.Decimal(0.00)
+        for order in orders:
+            amount = decimal.Decimal(order)
+            if order < 50:
+                disbursed_amount = disbursed_amount + amount + (amount * decimal.Decimal(0.01))
```

Comment:
You are adding the fees, however if we want to charge them to the merchant, is this the correct sign?